presto-main/src/main/java/com/facebook/presto/sql/planner/PlanFragmenter.java

```diff
@@ -282,6 +284,18 @@ private void chooseDistribution(PartitioningHandle distribution)
                     partitioningHandle);
         }
 
+        private boolean isCompatibleSystemPartitioning(PartitioningHandle distribution)
+        {
+            ConnectorPartitioningHandle currentHandle = partitioningHandle.get().getConnectorHandle();
+            ConnectorPartitioningHandle distributionHandle = distribution.getConnectorHandle();
+            if ((currentHandle instanceof SystemPartitioningHandle) &&
+                    (distributionHandle instanceof SystemPartitioningHandle)) {
+                return ((SystemPartitioningHandle) currentHandle).getPartitioning() ==
```

Comment:
As far as I understand it, this code just checks/makes sure that given `PlanFragment` can be scheduled and it checks that by checking input partitioning of a stage. It doesn't matter whether it is "fixed hash", "fixed arbitrary" or "fixed replicated" or  even "fixed replicate first row". It just must check  whether all of those could be handled by the same stage, which is true. 

On the other hand, two different source inputs couldn't be scheduled in one stage. As we discussed in different issue, one source input and one hash input might be possible to schedule in one stage, but are not supported yet.